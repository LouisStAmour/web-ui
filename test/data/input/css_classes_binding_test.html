<!doctype html>
<!--
Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
for details. All rights reserved. Use of this source code is governed by a
BSD-style license that can be found in the LICENSE file.
-->
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <script src="packages/web_ui/testing/testing.js"></script>
  <style>
  .uno    { width: 400px }
  .dos    { color: blue }
  .tres   { height: 50px }
  .cuatro { background-color: green }
  .cinco  { font-family: monospace  }
  </style>
</head>
<body>
<!-- test that you can set several class attributes with lists and strings with
spaces. -->
  <div id="test1" class="{{one}} {{two}} {{three}}">
  this div will have blue (0000ff) font color and 50px height.
  </div>
  <div id="test2" class="{{asList}}">
  this div will have blue 400px width and 50px height.
  </div>
  <div id="test3" class="{{asString}}">
  this div will have green (008000) bgcolor and monospace font.
  </div>
  <script type="application/dart">
    @observable
    library main;
    import 'dart:async';
    import 'dart:html';
    import 'package:web_ui/web_ui.dart';
    import 'package:unittest/unittest.dart';

    var one = 'uno';
    var two = 'dos';
    var three = 'tres';
    var asString = 'uno dos tres';
    var asList = toObservable(['uno', 'dos', 'tres']);
    main() {
      Timer.run(() {
        expect(query('#test1').classes, unorderedEquals(asList));
        expect(query('#test2').classes, unorderedEquals(asList));
        expect(query('#test3').classes, unorderedEquals(asList));
        one = null;
        asList[1] = null;
        asString = 'cuatro cinco';
        Timer.run(() {
          expect(query('#test1').classes, unorderedEquals(['dos', 'tres']));
          expect(query('#test2').classes, unorderedEquals(['uno', 'tres']));
          expect(query('#test3').classes, unorderedEquals(['cuatro', 'cinco']));
          window.postMessage('done', '*');
        });
      });
    }
  </script>
</body>
</html>
