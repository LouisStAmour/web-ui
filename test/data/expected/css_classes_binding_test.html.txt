Content-Type: text/plain
<html lang="en"><head><style>template { display: none; }</style>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <script src="../input/packages/web_ui/testing/testing.js"></script>
  <style>.uno {
  width: 400px;
}
.dos {
  color: #00f;
}
.tres {
  height: 50px;
}
.cuatro {
  background-color: #008000;
}
.cinco {
  font-family: monospace;
}</style>
</head>
<body>
<!-- test that you can set several class attributes with lists and strings with
spaces. -->
  <div id="test1" class="dos tres">
  this div will have blue (0000ff) font color and 50px height.
  </div>
  <div id="test2" class="uno tres">
  this div will have blue 400px width and 50px height.
  </div>
  <div id="test3" class="cuatro cinco">
  this div will have green (008000) bgcolor and monospace font.
  </div>
  


<script type="text/javascript" src="packages/web_ui/shadowdom.debug.js"></script>
<script type="application/dart" src="css_classes_binding_test.html_bootstrap.dart"></script></body></html>
